module/applications/journal/spell-list-page-sheet.mjs

```diff
@@ -0,0 +1,179 @@
+          section = context.sections[letter] ??= { header: letter.toUpperCase(), spells: [] };
+          break;
+        default:
+          continue;
+      }
+      section.spells.push(spell);
+    }
+    if ( context.grouping === "school" ) context.sections = sortObjectEntries(context.sections, "header");
+
+    return context;
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Load indices with necessary information for spells.
+   * @param {string} grouping  Grouping mode to respect.
+   * @returns {object[]}
+   */
+  async prepareSpells(grouping) {
+    let fields;
+    switch ( grouping ) {
+      case "level": fields = ["system.level"]; break;
+      case "school": fields = ["system.school"]; break;
+      default: fields = []; break;
+    }
+
+    let collections = new Collection();
+    for ( const uuid of this.document.system.spells ) {
+      const { collection } = foundry.utils.parseUuid(uuid);
+      if ( collection && !collections.has(collection) ) {
+        if ( collection instanceof Items5e ) collections.set(collection, collection);
+        else collections.set(collection, collection.getIndex({ fields }));
+      }
+    }
+
+    return (await Promise.all(collections.values()))
+      .flatMap(c => c.filter(s => this.document.system.spells.has(s.uuid)))
+      .sort((lhs, rhs) => lhs.name.localeCompare(rhs.name, game.i18n.lang));
+  }
+
+  /* -------------------------------------------- */
+  /*  Event Handlers                              */
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  activateListeners(jQuery) {
+    super.activateListeners(jQuery);
+    const [html] = jQuery;
+
+    html.querySelector('[name="grouping"]')?.addEventListener("change", event => {
+      this.grouping = (event.target.value === this.document.system.grouping) ? null : event.target.value;
+      this.object.parent.sheet.render();
+    });
+    html.querySelectorAll(".item-delete").forEach(e => {
+      e.addEventListener("click", this._onDeleteItem.bind(this));
+    });
+  }
+
+  /* -------------------------------------------- */
+
+  /**
+   * Handle deleting a dropped spell.
+   * @param {Event} event  This triggering click event.
+   * @returns {JournalSpellListPageSheet}
+   */
+  async _onDeleteItem(event) {
+    event.preventDefault();
+    const uuidToDelete = event.currentTarget.closest("[data-item-uuid]")?.dataset.itemUuid;
+    if ( !uuidToDelete ) return this;
+    const spellSet = this.document.system.spells.filter(s => s !== uuidToDelete);
+    await this.document.update({"system.spells": Array.from(spellSet)});
+    this.render();
+  }
+
+  /* -------------------------------------------- */
+
+  /** @inheritDoc */
+  async _onDrop(event) {
+    const data = TextEditor.getDragEventData(event);
+    let spells;
+    switch ( data?.type ) {
+      case "Folder":
+        spells = (await Folder.implementation.fromDropData(data))?.contents;
```

Comment:
I noticed this only retrieves the immediate children of the folder rather than any spells that are contained at some level under the dropped folder. I don't feel strongly on whether that should change but wanted to mention it in case you intended for the latter behaviour.

Reply:
This does match the current behavior of dropping a folder anywhere else which only handles one level, so I'd prefer to keep everywhere matching for the moment unless we decide to adjust other locations where we also accept dropped folders.